code/modules/mob/living/simple_animal/hostile/megafauna/dragon.dm

```diff
@@ -212,7 +212,7 @@ Difficulty: Medium
 		fire_rain()
 
 	icon_state = "dragon"
-	if(swoop_target && !qdeleted(swoop_target))
+	if(swoop_target && !qdeleted(swoop_target) && swoop_target.z == src.z)
```

Comment:
REEEEEEEE FINE BUT IT HONESTLY MAKES IT MORE READABLE THAN SOME RANDOM Z THAT LOOKS LIKE A TYPO